cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,138 @@
+		assert_ok!(<RococoRelay as RococoRelayPallet>::Identity::set_identity(
+			rococo_runtime::RuntimeOrigin::signed(RococoRelaySender::get()),
+			Box::new(identity_relaychain)
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Identity(pallet_identity::Event::IdentitySet { .. }) => {},
+			]
+		);
+
+		// 2. Set sub-identity on Relay Chain
+		assert_ok!(<RococoRelay as RococoRelayPallet>::Identity::set_subs(
+			rococo_runtime::RuntimeOrigin::signed(RococoRelaySender::get()),
+			vec![(RococoRelaySender::get(), Data::Raw(vec![40; 1].try_into().unwrap()))],
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Balances(pallet_balances::Event::Reserved { .. }) => {},
+				RuntimeEvent::Identity(pallet_identity::Event::IdentitySet { .. }) => {},
+				RuntimeEvent::Balances(pallet_balances::Event::Reserved { .. }) => {},
+			]
+		);
+
+		bal_after_relaychain =
+			<RococoRelay as RococoRelayPallet>::Balances::free_balance(RococoRelaySender::get());
+	});
+
+	// Set identity and Subs on Parachain with Zero deposit
+	PeopleRococo::execute_with(|| {
+		type RuntimeEvent = <PeopleRococo as Chain>::RuntimeEvent;
+
+		bal_before_parachain =
+			<PeopleRococo as PeopleRococoPallet>::Balances::free_balance(PeopleRococoSender::get());
+
+		// 3. Set identity on Parachain with zero deposit
+		assert_ok!(<PeopleRococo as PeopleRococoPallet>::Identity::set_identity_no_deposit(
+			&PeopleRococoSender::get(),
+			identity_parachain.clone()
+		));
+
+		// 4. Set sub-identity on Parachain
+		assert_ok!(<PeopleRococo as PeopleRococoPallet>::Identity::set_subs(
+			people_rococo_runtime::RuntimeOrigin::signed(PeopleRococoSender::get()),
+			vec![(PeopleRococoSender::get(), Data::Raw(vec![40; 1].try_into().unwrap()))],
+		));
+		assert_expected_events!(
+			PeopleRococo,
+			vec![
+				RuntimeEvent::Balances(pallet_balances::Event::Reserved { .. }) => {},
+			]
+		);
+
+		bal_after_parachain =
+			<PeopleRococo as PeopleRococoPallet>::Balances::free_balance(PeopleRococoSender::get());
+		let fees = bal_before_parachain - bal_after_parachain;
+		assert_eq!(bal_before_parachain - fees, bal_after_parachain);
+	});
+
+	// 5. reap_identity on Relay Chain
+	RococoRelay::execute_with(|| {
+		type RuntimeEvent = <RococoRelay as Chain>::RuntimeEvent;
+		assert_ok!(<RococoRelay as RococoRelayPallet>::Identity::reap_identity(
+			&RococoRelaySender::get(),
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Balances(pallet_balances::Event::Unreserved { .. }) => {},
+			]
+		);
+		assert!(<RococoRelay as RococoRelayPallet>::Identity::identity(&RococoRelaySender::get())
+			.is_none());
+		assert!(<RococoRelay as RococoRelayPallet>::Identity::super_of(&RococoRelaySender::get())
+			.is_none());
+		let bal_after_reap =
+			<RococoRelay as RococoRelayPallet>::Balances::free_balance(RococoRelaySender::get());
+		assert_eq!(bal_after_reap, bal_before_relaychain)
+	});
```

Comment:
Missing more assertions. This needs to enter another parachain block (advancing the block height and processing the message queue). It then needs to confirm:

- Asset was teleported
- Identity deposit was updated to the correct amount
- `DepositUpdated` event
- A single existential deposit amount is free

Reply:
Other than the event assertion, the other asserts have been added. 

I'm not able to assert on the `DepositUpdated` event, because when we call the `construct_runtimes!` macro in `people-rococo/src/lib.rs`:

On L1398 we do:
`IdentityMigrator: identity_migrator ...`

It seems we can only assert on Event enums when fields declared in this macro use the name `pallet_`. For example:
`Identity: pallet_identity`

I did try updating `IdentityMigrator` to: 
`IdentityMigrator: pallet_identity_migrator`, but this cause the Runtime to not build correctly. This could be good reason for an issue to update that macro so that we can assert on these non-pallet-named enums.

Reply:
Here https://github.com/paritytech/polkadot-sdk/blob/7a29719f2376ce83a6f5ea5e3a85b4d3d8967707/cumulus/parachains/runtimes/people/people-rococo/src/lib.rs#L449

Reply:
Yeah I know the pallet name, I wrote it :P. I mean the requirement that it must begin with `pallet_`.

Reply:
Where we implement syn `Parse` trait, but I need to dig deeper to find the exact cause of the bug. https://github.com/paritytech/polkadot-sdk/blob/21f1811c6600d8a7fe043592ff34dcb79284d583/substrate/frame/support/procedural/src/construct_runtime/parse.rs#L82C3-L82C3

Reply:
That doesn't seem right, I mean this mock runtime has tons of non-`pallet_` pallets.
https://github.com/paritytech/polkadot-sdk/blob/35eb133/polkadot/runtime/parachains/src/mock.rs#L53

Reply:
Managed to bring it into scope and assert by doing: `polkadot_runtime_common::identity_migrator::Event::DepositUpdated` 🥳